crates/accelerate/src/sabre/route.rs

```diff
@@ -573,14 +600,16 @@ pub fn swap_map_trial(
         }
         if routable_nodes.is_empty() {
             // If we exceeded the max number of heuristic-chosen swaps without making progress,
-            // unwind to the last progress point and greedily swap to bring a ndoe together.
+            // unwind to the last progress point and greedily swap to bring a node together.
             // Efficiency doesn't matter much; this path never gets taken unless we're unlucky.
             current_swaps
                 .drain(..)
                 .rev()
                 .for_each(|swap| state.apply_swap(swap));
             let force_routed = state.force_enable_closest_node(&mut current_swaps);
-            routable_nodes.push(force_routed);
+            force_routed
+                .iter()
+                .for_each(|routable_node| routable_nodes.push(*routable_node));
```

Comment:
```suggestion
            routable_nodes.extend(force_routed);
```
Maybe just this?

(Also, have your Rust travels transcended you beyond the use of the 'for' loop? 🤣)

Reply:
Lol yes, Rust has me thinking in one-liners where possible 😂